src/client/refresh.js

```diff
@@ -12,9 +12,16 @@ export default function _refresh (options = {}) {
    *
    * @return {Object} - new meta info
    */
-  return function refresh () {
-    const info = getMetaInfo(options)(this.$root)
-    updateClientMetaInfo(options).call(this, info)
-    return info
+  return function refresh() {
+    const metaInfo = getMetaInfo(options, this.$root)
+
+    const tags = updateClientMetaInfo(options, metaInfo)
+
+    // emit "event" with new info
+    if (tags && typeof metaInfo.changed === 'function') {
+      metaInfo.changed.call(this, metaInfo, tags.addedTags, tags.removedTags)
```

Comment:
The docs state that `changed` is called with a `this` context of the component on which the `changed` function is defined. This doesnt seem to be the case, refresh is/was called for the component which triggers the meta update and that component is expicitly set as the `this` context. Because of that, when there are multiple changes in a row the `changed` function will only be called for the last component that called refresh (due to the batchUpdate) and will have the last component that changed as the this context.

Maybe I broke this behaviour with the code refactor by removing all the function wrapping it used, but I dont think so. See my last commit to verify this behaviour: https://github.com/nuxt/vue-meta/pull/319/commits/3cfbf878bbae2e5c2820fd27eec4a872ac241b1d

Reply:
/cc @Atinux

Reply:
Well, I never used this function actually, but I believe this PR will be for next major version.